torch_geometric/explain/algorithm/gnn_explainer.py

```diff
@@ -201,19 +201,30 @@ def _initialize_masks(self, x: Tensor, edge_index: Tensor):
     def _loss_regression(self, y_hat: Tensor, y: Tensor) -> Tensor:
         return F.mse_loss(y_hat, y)
 
-    def _loss_classification(self, y_hat: Tensor, y: Tensor) -> Tensor:
+    def _loss_multiclass_classification(self, y_hat: Tensor,
+                                        y: Tensor) -> Tensor:
         if y.dim() == 0:  # `index` was given as an integer.
             y_hat, y = y_hat.unsqueeze(0), y.unsqueeze(0)
 
         y_hat = self._to_log_prob(y_hat)
 
         return (-y_hat).gather(1, y.view(-1, 1)).mean()
 
+    def _loss_binary_classification(self, y_hat: Tensor, y: Tensor) -> Tensor:
+        if self.model_config.return_type.value == 'raw':
+            return F.binary_cross_entropy_with_logits(y_hat, y.float())
```

Comment:
I like the reuse here of `F.binary_cross_entropy_with_logits`. We should probably do the same for `_loss_multiclass_classification` as well and use `F.cross_entropy_with_logits`.

Reply:
do you mean something like
```python
    def _loss_multiclass_classification(self, y_hat: Tensor,
                                        y: Tensor) -> Tensor:
        if y.dim() == 0:  # `index` was given as an integer.
            y_hat, y = y_hat.unsqueeze(0), y.unsqueeze(0)

        if self.model_config.return_type.value == 'raw':
            return F.cross_entropy_with_logits(y_hat.view_as(y), y.float())
        if self.model_config.return_type.value == 'probs':
            return F.cross_entropy(y_hat.view_as(y), y.float())
        if self.model_config.return_type.value == 'log_probs':
            return (-y_hat).gather(1, y.view(-1, 1)).mean()
```
if yes, the `raw` and `probs` cases are clearer but we don't get rid of the hard to decode `log_probs` case.
Maybe you have something simpler in mind?